frontend/src/app/main/ui/workspace/sidebar/options/menus/color_selection.cljs

```diff
@@ -121,49 +129,40 @@
      (when open?
        [:div {:class (stl/css :element-content)}
         [:div {:class (stl/css :selected-color-group)}
-         (for [[index color] (d/enumerate (take 3 library-colors))]
-           [:& color-row {:key (dm/str "library-color-" index)
-                          :color color
-                          :index index
-                          :on-detach on-detach
-                          :select-only select-only
-                          :on-change #(on-change %1 color %2)
-                          :on-open on-open
-                          :on-close on-close}])
+         ;; The hidden color is to solve a problem with the color picker. When a color is changed
+         ;; and is no longer a library color it disapears from the list of library colors. Because
+         ;; we need to keep the color picker open we need to maintain that color. The easier way
+         ;; is to render the color elements so even if the library color is no longer we have still
+         ;; the component to change it from the color picker.
+         (let [lib-colors (concat (cond->> library-colors (not @expand-lib-color) (take 3)) colors)]
```

Comment:
The `concat` on this line looks unnecessary, because you only passing a single expression to it.